src/lib.rs

```diff
@@ -151,8 +151,5 @@ mod loop_logic;
 mod sources;
 
 fn no_nix_err(err: nix::Error) -> std::io::Error {
-    match err {
-        ::nix::Error::Sys(errno) => errno.into(),
-        _ => unreachable!(),
-    }
+    err.into()
 }
```

Comment:
If there is now an `Into<std::io::Error>` impl for `nix::Errno`, then this `no_nix_err` function should just be removed, as it no longer serves any purpose.

Reply:
Keeping this function helps with type inference in several places (where we call `.map_err(no_nix_err)`)

Reply:
I'd prefer if those places where properly fixed rather than keeping `no_nix_err` as a mere shim around `Into::into()`.